wordpress-mobile/WordPress-iOS
# [Modal Layout Picker] Request scaled Thumbnail images

**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2445

## To test:
<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

🗒️ More persistent caching of the thumbnail will be handled by a later ticket this currently uses an in-memory cache that will be purged by going to the background.
🗒️ If multiple image downloads complete at the same time the app can start to feel choppy. I plan to revisit this when we handle the long term cache of thumbnails

### WPCom Site
1. Select a WPcom site and ensure a valid network connection
1. Navigate to the Modal layout picker by either:
    - Selecting the FAB icon on the site page and selecting "Site page"
    - Selecting the ➕ on the Page List
1. Using a network logger:
    - **Expect** to see the `preview` parameter return with data like: 
        ```
        "preview": "https:\/\/headstartdata.files.wordpress.com\/2020\/01\/about-3.png?w=828&zoom=2"
        ```
    - **Expect** to see a network request with: 
        ```
        https://public-api.wordpress.com/wpcom/v2/sites/{{SITEID}}/block-layouts?locale=en&preview_width=828.0&scale=2.0
        ```

### Self-Hosted Site
1. Select a Self-Hosted site and ensure a valid network connection
1. Navigate to the Modal layout picker by either:
    - Selecting the FAB icon on the site page and selecting "Site page"
    - Selecting the ➕ on the Page List
1. Using a network logger:
    - **Expect** to see the `preview` parameter return with data like: 
        ```
        "preview": "https:\/\/headstartdata.files.wordpress.com\/2020\/01\/about-3.png?w=828&zoom=2"
        ```
    - **Expect** to see a network request with: 
        ```
        https://public-api.wordpress.com/wpcom/v2/common-block-layouts?locale=en&preview_width=828.0&scale=2.0
        ```

## Screenshots:
<img width="300" alt="Screen Shot 2020-08-27 at 5 17 58 PM" src="https://user-images.githubusercontent.com/3384451/91495909-4a200f00-e889-11ea-959d-c7b05a09a1b2.png">

## Additional Context: 

This is a continuation of the work done in https://github.com/wordpress-mobile/WordPress-iOS/pull/14726 and will be followed up by other tickets as part of [Modal Layout Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/issues?q=is%3Aissue+is%3Aopen+%5BModal+Layout+Picker%5D)

## PR submission checklist:
